wandb/sdk/internal/sender.py

```diff
@@ -1449,6 +1449,30 @@ def send_request_log_artifact(self, record: "Record") -> None:
 
         self._respond_result(result)
 
+    def send_request_download_artifact(self, record: "Record") -> None:
+        # assert record.control.req_resp
+        result = proto_util._result_from_record(record)
+        qualified_name = record.request.download_artifact.qualified_name
+        download_root = record.request.download_artifact.download_root
+        recursive = record.request.download_artifact.recursive
+        allow_missing_references = (
+            record.request.download_artifact.allow_missing_references
+        )
+
+        try:
+            artifact = wandb.Api().artifact(qualified_name)
```

Comment:
This is brittle (see e.g. [here](https://weightsandbiases.slack.com/archives/C02DTSCHND8/p1695936099975289)), but I can't think of a better way...